src/utils.c

```diff
@@ -49,6 +49,37 @@
 
 extern char *aux_resource_path;
 
+#ifdef USE_BZIP2
+
+#include <stdarg.h>
+#include <bzlib.h>
+
+/* This function is an API glue between zprintf and
+ * the subsequent bzwrite to a compressed file. BZ2
+ * doesn't have a printf interface, so an equivalent
+ * interface needed to be written.
+ * Note: this function can only output 512 chars at a time.
+ *   if a string comes through bigger than 512, it will be
+ *   truncated. Perhaps this can be optimized in the future.
+ *
+ */
+#define BZ_MAX_SIZE 512
```

Comment:
Do you think 512 is large enough for all of the data fields that Joy currently has? It should be fine with large data (i.e. IDP) that is printed with zprintf_raw_as_hex because that loops through the data printing a hex character at a time.